### PR TITLE
Add spiutils lib.

### PR DESCRIPTION
**Pull request by osenft**
This library will be used both by Tock kernel and in userspace.

### CODE REVIEW

File: shared-lib/spiutils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+
```
Comment (jrvanwhy) [inline shared-lib/spiutils/Cargo.toml]: Please remove the blank line after `[package]`.

File: shared-lib/spiutils/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+name = "spiutils"
+version = "0.1.0"
+authors = [ "Oskar Senft <osk@google.com>" ]
+license = "MIT"
```
Comment (jrvanwhy) [inline shared-lib/spiutils/Cargo.toml]: This is `Apache-2.0` licensed, not `MIT` licensed.

File: shared-lib/spiutils/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+keywords = ["spi-utils"]
```
Comment (jrvanwhy) [inline shared-lib/spiutils/Cargo.toml]: `spi-utils` isn't really a good keyword. I don't expect this to be uploaded to crates.io (do you?) so I don't think we need to specify keywords.

File: shared-lib/spiutils/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+description = """
+Utilities for SPI
```
Comment (jrvanwhy) [inline shared-lib/spiutils/Cargo.toml:7]: I'm not sure this description is very good -- this crate is probably going to be somewhat Titan-specific, correct?

I think we can remove it, unless you expect this to be uploaded to `crates.io`.

File: shared-lib/spiutils/src/cmds.rs [hunk 5]
```diff
@@ -0,0 +1,86 @@
+/// SPI flash op codes
+#[repr(u8)]
+#[allow(dead_code)]
+pub enum OpCodes {
+    /// No operation
+    Nop = 0x00,
+
+
```
Comment (jrvanwhy) [inline shared-lib/spiutils/src/cmds.rs:9]: Too many blank lines here? Probably should just be 1.

File: shared-lib/spiutils/src/cmds.rs [hunk 6]
```diff
@@ -0,0 +1,86 @@
+/// SPI flash op codes
+#[repr(u8)]
+#[allow(dead_code)]
```
Comment (jrvanwhy) [inline shared-lib/spiutils/src/cmds.rs]: `#[allow(dead_code)]` shouldn't be necessary here.

**Review by jrvanwhy — CHANGES_REQUESTED** [review]

File: shared-lib/spiutils/Cargo.toml (hunk 4 above)
Comment (osenft) [reply shared-lib/spiutils/Cargo.toml:7]: I was expecting the crate to be independent from Titan. Two things I know I want in there:
- SPI flash commands (done)
- SPI flash ID generation like JEDEC ID and SFDP.

File: shared-lib/spiutils/Cargo.toml (hunk 1 above)
Comment (osenft) [reply shared-lib/spiutils/Cargo.toml]: Done.

File: shared-lib/spiutils/Cargo.toml (hunk 2 above)
Comment (osenft) [reply shared-lib/spiutils/Cargo.toml]: Changed. I wasn't sure what this should be set to, I think I copied it from another lib. Thanks for clarifying!

File: shared-lib/spiutils/Cargo.toml (hunk 4 above)
Comment (jrvanwhy) [reply shared-lib/spiutils/Cargo.toml:7]: Perhaps the crate name should be something like `spi_flash_decls`? `spiutils` isn't descriptive enough for `crates.io`.

File: shared-lib/spiutils/src/cmds.rs (hunk 5 above)
Comment (osenft) [reply shared-lib/spiutils/src/cmds.rs:9]: I was using that to visually distinguish commands from each other. I added some additional comment in.

File: shared-lib/spiutils/src/cmds.rs (hunk 6 above)
Comment (osenft) [reply shared-lib/spiutils/src/cmds.rs]: Ah yes, good catch, thank you.

File: shared-lib/spiutils/Cargo.toml (hunk 3 above)
Comment (osenft) [reply shared-lib/spiutils/Cargo.toml]: Removed.

File: shared-lib/spiutils/Cargo.toml (hunk 4 above)
Comment (jrvanwhy) [reply shared-lib/spiutils/Cargo.toml:7]: Do you want to give this a name more suitable for `crates.io` or submit as-is?

File: shared-lib/spiutils/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "spiutils"
+version = "0.1.0"
+authors = [ "Oskar Senft <osk@google.com>" ]
+license = "Apache-2.0"
+description = """
+Utilities for SPI
+"""
```
Comment (jrvanwhy) [inline shared-lib/spiutils/Cargo.toml:8]: Cargo.toml is missing an EOL character at its end.

**Review by jrvanwhy — APPROVED** [review]
Please replace the missing EOL character in `Cargo.toml`, otherwise LGTM.

**Pull request merged**